core/domain/skill_domain.py

```diff
@@ -698,6 +727,7 @@ def add_misconception(self, misconception_dict):
         """
         misconception = Misconception(
             misconception_dict['id'],
+            misconception_dict['skill_id'],
```

Comment:
I'd suggest having the skill_id as a separate argument for add_misconception.
 Since, this change would mean the frontend has to be changed, which is not really required, as the calling function would know what skill the misconception belongs to.
Or, if you are going ahead with this, add a validation check in skill.validate and here, to make sure all misconceptions in skill have the correct skill_id.

Reply:
If I separate skill_id from the misconception_dict, then it wouldn't be a complete dict, which would cause some misunderstanding. I think I should go with the second solution even though it's a little more complicated.

Please correct me if I'm wrong!

Reply:
This confuses me a bit, I have limited knowledge in Misconceptions so correct me if I am wrong - the misconceptions dict is part of a skill (which has its id) right? So why are we putting a skill_id within the misconception dict as well ? I am guessing we are storing it this way in the backend too? 
To me that seems like unnecessary redundancy - Because we have a Skill model with its id, within that a misconception and within that again a skill_id.
When we fetch misconceptions from the datastore we anyway have to fetch it from skills from what I understand right?

Reply:
One purpose was to fix some errors caused by multiple skills linked to one question. Currently, we can't distinguish two misconceptions with the same id but belong to different skills in the frontend.
I also remember we talked about adding a misconception skill link for the use of skillMasteryLevel, but I thought that might be too much work, so I suggested doing it in this way.

Reply:
Normally we don't store this id in other objects like skillSummary or skillContent because their id is the same as the skill_id, so we can check which skill it belongs to very easily, but for misconceptions, their ids are just 0, 1, 2..., depending on the order they are created.

Reply:
I didn't completely understand how adding the skill_id within the misconception in the backend is fixing the multiple skills issue..from my understanding, that is fixed by now using tagged_misconception_skill_id no? 
Like you said, you can store the misconception as a dictionary, that sounds good to me.

Do we still need the misconception skill link anymore? I think we just to figure a link from answer groups to misconceptions, and since the tagged_skill_misconception_id anyway has skill_id in it, we don't need skill_id in misconceptions (atleast from what I understand)
But I will defer to you and @aks681 as to what is the right way to do things because I am not really aware about Misconceptions much!

Reply:
Reverted.